tavern/core.py

```diff
@@ -155,7 +158,32 @@ def run_stage(sessions, stage, tavern_box, test_block_config):
     delay(stage, "after")
 
 
-def run(in_file, tavern_global_cfg, tavern_http_backend, tavern_mqtt_backend, tavern_strict):
+def _run_pytest(in_file, pytest_args=None, **kwargs):
+    """Run all tests contained in a file using pytest.main()
+
+    Args:
+        in_file (str): file to run tests on
+        ptest_args (list): Extra pytest args to pass
+
+        **kwargs (dict): ignored
+
+    Returns:
+        bool: Whether ALL tests passed or not
+    """
+
+    if kwargs:
+        warnings.warn("Passing extra keyword args to run() when using pytest is used are ignored.", FutureWarning)
+
+    if pytest_args is None:
+        import sys
+        sys.argv += ["-k", in_file]
+    else:
+        pytest_args += ["-k", in_file]
+
+    return pytest.main(args=pytest_args)
```

Comment:
Is it possible to override the test collection in order to reduce the time penalty for this?

Reply:
If we can do that, I suggest that we drop the `run_legacy` straight away because it's another thing to maintain.

Reply:
agreed